2024/planning-process.md

```diff
@@ -0,0 +1,187 @@
+        2. [Khronos Group](https://www.khronos.org/)
+        3. [TC39](https://tc39.es/) (Stage 2+)
+        4. [W3C](https://www.w3.org/) (Recommendation Track)
+        5. [WHATWG](https://whatwg.org/)
+    3. Describe the benefit to web developers and the web itself
+    4. Link to existing tests (if any) and/or describe plan to create needed tests
+5. Investigation Project proposals must:
+    1. Outline their scope and their goals,
+    2. Not substantially be work better suited to a group within a standards development organization,
+    3. Clearly provide present or future impact on cross-browser interoperability.
+
+As soon as each proposal is submitted as a GitHub issue, discussion of that proposal can begin. Questions can be asked. The proposal can be edited and refined.
+
+At the end of the three weeks, the deadline will end the period when brand new proposals can be submitted.
+
+A fourth week will allow for continued discussion and refinement. The deadline at the end of the fourth week ends the opportunity to significantly change the scope of the proposal.  The proposal will be evaluated on its state at that time.
+
+
+## Focus Area Selection
+
+### Elimination Round 1 : Disqualification
+
+After the Call of Proposals Period has ended, the organizing committee will finalize an official list of all Focus Area proposals.
+
+Any Focus Area proposal that does not meet the minimum publicly-stated selection criteria will be considered for preliminary elimination. For example:
+
+1. The proposal is overly vague or broad.
+2. The proposed technology is already interoperable according to WPT results.
+3. The proposed technology is not on a standards track, or its standard is not sufficiently mature.
+
+Decisions will be made at a specific meeting for discussion of preliminary eliminations. Any member organization can bring up a proposal they believe should be eliminated. Brief debate can be had. If **there is quick consensus**, that proposal will be eliminated. If there is not consensus and a longer debate is needed, the proposal will be not be eliminated and will instead move on to the next round.
+
+1. Setup *Disqualification meeting*
+2. Each organization can propose items for disqualification.
+3. Brief debate on each proposal:
+    1. if there is quick consensus, disqualify the proposal
+    2. otherwise, send the proposal to Round 2
+4. Publish the list of proposals pushed to Round 2.
+
+### Elimination Round 2: Reduction
+
+At this point, there is a clear list of Focus Area proposals under consideration. Each organization has two weeks to review the list of proposals and make Round 2 decisions.
+
+Each member organization chooses `N` number of proposals they would like to see move on to Round 3. There is no ranking or scoring at this point, a simple “yes, we want this proposal to stay in consideration”. Once all the “yeses” are gathered, any proposal that does not have the support of at least one organization is eliminated.
```

Comment:
I don't understand how this will work in detail.

Incoming proposals may be different sizes. Per the above text some proposals may be far too small to form a focus area on their own. So in order to do fixed-N selection I think we'd need to have already grouped the proposals into similar sized chunks at this point. Otherwise it's impossible to compare proposals like "fix one small but important bug" and e.g. "implement an entire new CSS layout mode". 

Also it's unclear how to pick N. We probably need N large enough that we're guaranteed to get enough proposals that Interop 2024 is a reasonable sized project. But given overlaps and possible later vetos, that doesn't seem possible without allowing the process to be iterative. One point of view is "well even if we only end up with five focus areas, we'll know those are the most important ones". But I think that would be a mistake if we end up rejecting things that everyone does in fact end up working on just because Process.

Reply:
N should be "slightly too large". 

If you imagine we expect 35 proposals and 20 "open slots", then figuring out N does seem impossible. And honestly, this whole step could be skipped. Just take all 35 proposals to the next round. 

But if you instead imagine we could get 150 proposals and have only 5 open slots... then this step makes more sense. It's a way to make a huge cut very quickly to get the number of proposals into a reasonable number. Why sweat the details over which is ranked 82nd and which is ranked 97th if we are only accepting 9 new areas? 

We won't know how many open slots there are, but we could guess, for instance, it's 15 at most, as so... let's make N be... 25? Or 40? And cut the other 100+ proposals (if that many come in).

> We probably need N large enough that we're guaranteed to get enough proposals that Interop 2024 is a reasonable sized project. 

Exactly. And it's fine if N is "too high". It's really just a protection against have way too many proposals in the later stage of the process, just in case the success of and attention on Interop 2021-23 brings us exponentially more proposals yet again this year.

Reply:
+1 to jgraham's comment above about selecting 'N' being particularly tricky. Suggest each organization pick a reasonable number of proposals that are well-justified and that they can support.

Reply:
One of the big questions that came up in the meetings in recent weeks was the question of when to do the grouping; it might be worth revisiting this after deciding when to do the grouping?

Reply:
How to do any grouping, and how to estimate/balance the overall size of the focus areas, seem like the most fundamental remaining issues. Previously we did that at the last minute by "feel", but that doesn't work well with the process set out here due to the ordering step.

 Grouping before ordering doesn't work if that happens before the proposals have got detailed input from engineering teams, because we might group things that have very different levels of complexity or priority, and then be forced to assess the group as a whole on the basis of its worst elements.
 
 But trying to order things of very different size and complexity also doesn't seem to work well (how does one tradeoff "fix a single test that represents a serious webcompat issue but is probably a few days of work and isn't going to be a focus area on its own" against "implement an entire new platform feature that could take multiple months of work"?).

Reply:
The suggestion I like the best so far is this one from @jgraham:

> ask people to pick a number of proposals that they feel reflects a reasonable size for Interop 2024

I think there's general agreement that we should not increase the number of focus areas, so at the very very most this could be 26.

This does require each organization to make some qualified guesses about what will make sense to group after round 3. In the case of web compat bugs this is easy. For some other proposals there might be disagreement, but the only effect of this is that a few more proposals move on to round 3.

Corner case consideration: Elsewhere I suggested we should also allow for vetos in round 2. If we do this, then there's no guarantee that we have N proposals after round 2. I think this is very unlikely, but if it does happen we could do another round of round 2 for the remaining slots. Presumably there would be no vetos that time.